### PR TITLE
Swapped kinesis opencensus producer with omnition producer

### PR DESCRIPTION
**Pull request by Excaleo**
**Description:** 
* Swapped kinesis opencensus producer with omnition producer

**Testing:**
* Changed `config.yaml` under `kinesisexporter/testdata` to match the new configuration parameters

### CODE REVIEW

File: exporter/kinesisexporter/factory.go [hunk 1]
```diff
@@ -68,32 +67,25 @@ func createTraceExporter(
 	config configmodels.Exporter,
 ) (component.TraceExporter, error) {
 	c := config.(*Config)
-	k, err := kinesis.NewExporter(&kinesis.Options{
-		Name:               c.Name(),
-		StreamName:         c.AWS.StreamName,
-		AWSRegion:          c.AWS.Region,
-		AWSRole:            c.AWS.Role,
-		AWSKinesisEndpoint: c.AWS.KinesisEndpoint,
+	awsConfig := aws.NewConfig().WithRegion(c.AWS.Region).WithEndpoint(c.AWS.KinesisEndpoint)
```
Comment (irisgve) [inline exporter/kinesisexporter/factory.go:70]: We might want to move this within `exporter.go` with a new exporter function to abstract the AWS/Kinesis setups out of the entrypoint.

Comment (irisgve) [reply exporter/kinesisexporter/factory.go:70]: That could also include setting this up: https://godoc.org/github.com/a8m/kinesis-producer#Producer.NotifyFailures

I just realized that https://github.com/signalfx/omnition-kinesis-producer is just a fork of https://godoc.org/github.com/a8m/kinesis-producer

File: exporter/kinesisexporter/config_test.go [hunk 2]
```diff
@@ -56,20 +56,13 @@ func TestDefaultConfig(t *testing.T) {
 				FlushIntervalSeconds: 5,
 				MaxConnections:       24,
 			},
-
```
Comment (irisgve) [inline exporter/kinesisexporter/config_test.go:59]: Looking at this now, I wonder what we lose when we move to the omnition library. There seems to be some sharding logic that the opencensus-go-exporter is doing before sending it thru to the streams. Currently checking what's it doing.

File: exporter/kinesisexporter/factory.go (hunk 1 above)
Comment (Excaleo) [reply exporter/kinesisexporter/factory.go:70]: Will create a kinesis exporter instantiation function in  `exporter.go`

File: exporter/kinesisexporter/config_test.go (hunk 2 above)
Comment (Excaleo) [reply exporter/kinesisexporter/config_test.go:59]: The sharding/paritition key logic in opencensus was done using the span ID, I replicated the behaviour within the ConsumeTraces function. The other parameters here seem to be parameters of the producer itself. 

I think we sacrifice fine grained control of the external library by switching to this.

Comment (irisgve) [reply exporter/kinesisexporter/config_test.go:59]: Yeah, I guess we could revisit whatever `shardProducer` logic they have once we're ready to submit this upstream. But for our current basic case, I'm good with it.

**Review by irisgve — APPROVED** [review]

**Pull request merged**